common/djangoapps/third_party_auth/models.py

```diff
@@ -827,6 +829,41 @@ def get_setting(self, name):
             return other_settings[name]
         raise KeyError
 
+    def get_current_saml_configuration(self):
+        """
+        Get the current active SAMLConfiguration for this provider.
+        Falls back to the default configuration if none is set or if the
+        current one is outdated.
+        """
+        if self.saml_configuration:
+            # Check if we have the current version
+            try:
+                current_config = SAMLConfiguration.current(self.site_id, self.saml_configuration.slug)
+                if current_config and current_config.id != self.saml_configuration_id:
```

Comment:
1. Do you mean `self.saml_configuration.id` and not `self.saml_configuration_id`? This makes me think that this code is not even tested locally with some manual testing, right?
2. In theory we won't need this update functionality once the management command is used and the receiver is in use. I'm wondering if we should simplify the code by leaving out this 3rd place where the records could be updated?